### ISSUE TITLE
add support for more native ffmpeg quality metrics

### ISSUE THREAD

**Issue by richardpl** [slhck/ffmpeg-quality-metrics#31]
There is now also vif (Visual Information Fidelity), and soon identity and msad (Mean Sum of Squared Differences).

They also supports slice threading and timeline (calculating quality metrics only for some frame ranges),

**Comment by slhck** [comment]
Thanks for the info! I'll probably add an option to calculate them in addition ot the default ones, like for VMAF.

**Comment by slhck** [comment]
VIF is now implemented since version 2.0.

**Comment by richardpl** [comment]
Cool.

**Comment by ls-milkyway** [comment]
Now ffmpeg ...i guess also supports dssim & ms_ssim

**Comment by slhck** [comment]
@ls-milkyway Why do you think FFmpeg supports DSSIM? Note that VMAF already includes an MS-SSIM computation.

**Comment by ls-milkyway** [comment]
> @ls-milkyway Why do you think FFmpeg supports DSSIM? Note that VMAF already includes an MS-SSIM computation.

I mean you can [now build ffmpeg with DSSIM](https://github.com/m-ab-s/media-autobuild_suite)...yep MS-SSIM is included example: `ffmpeg -i distorted.mpg -i reference.mpg -lavfi libvmaf='feature=name=ms_ssim' -f null -`

**Comment by ls-milkyway** [comment]
**Please Note:** You can add/remove support for more native ffmpeg quality metrics...but **please do not change the JSON output structure** as I am using it extensively as an input to another program....thanks

**Comment by slhck** [comment]
> > @ls-milkyway Why do you think FFmpeg supports DSSIM? Note that VMAF already includes an MS-SSIM computation.
> 
> I mean you can [now build ffmpeg with DSSIM](https://github.com/m-ab-s/media-autobuild_suite)...yep MS-SSIM is included; example: `ffmpeg -i distorted.mpg -i reference.mpg -lavfi libvmaf='feature=name=ms_ssim' -f null -`

That seems to be a [third party addon](https://github.com/m-ab-s/media-autobuild_suite/blob/b984784516ea25b925cbd38757681d659be80681/build/media-suite_compile.sh#L211-L218) not provided by FFmpeg itself.

> **Please Note:** You can add/remove support for more native ffmpeg quality metrics...but **please do not change the JSON output structure** as I am using it extensively as an input to another program....thanks

Sure, I have experienced such problems myself and I'll promise not to change it! I will get back to this issue (and a few other VMAF-related problems) later in May.

**Comment by ls-milkyway** [comment]
> That seems to be a [third party addon](https://github.com/m-ab-s/media-autobuild_suite/blob/b984784516ea25b925cbd38757681d659be80681/build/media-suite_compile.sh#L211-L218) not provided by FFmpeg itself.

Yep, but now I guess [MS-SSIM is better evaluation than SSIM OR DSSIM](https://videoclarity.com/PDF/Advancing-To-Multi-Scale-SSIM-Final.pdf)...hence it's better to include MS-SSIM instead of SSIM OR DSSIM.

**Comment by slhck** [comment]
You can get MS-SSIM through VMAF (https://github.com/Netflix/vmaf/blob/master/libvmaf/src/feature/ms_ssim.c), so it should be possible to enable it. I will have to add support to map the respective CLI options though.

**Comment by slhck** [comment]
Note that MS-SSIM evaluation is now possible starting with version 3.x of this program. See the `--vmaf-features` option as explained in the README.

**Comment by slhck** [comment]
Added MSAD support in version 3.5.0.
